src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,14 +1,26 @@
 import React from 'react';
+import { Status } from '../../types/Status';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { setQuery, setStatus } from '../../features/filter';
 
 export const TodoFilter: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const filter = useAppSelector(state => state.filter);
+
+  const handleStatus = (event: React.ChangeEvent<HTMLSelectElement>) =>
+    dispatch(setStatus(event.target.value as Status));
```

Comment:
The `setStatus` action expects a value of type `Status`. Ensure that the `event.target.value` is a valid `Status` value. Consider adding validation or type checking to prevent runtime errors.